src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -2270,6 +2290,34 @@ bool ConditionMgr::isConditionTypeValid(Condition* cond) const
                 return false;
             }
             break;
+        case CONDITION_QUEST_OBJECTIVE_PROGRESS:
+        {
+            const Quest* quest = sObjectMgr->GetQuestTemplate(cond->ConditionValue1);
+            if (!quest)
+            {
+                TC_LOG_ERROR("sql.sql", "%s points to non-existing quest (%u), skipped.", cond->ToString(true).c_str(), cond->ConditionValue1);
+                return false;
+            }
+
+            if (cond->ConditionValue2 < 0 || cond->ConditionValue2 > 3)
```

Comment:
Could change it to handle values between 1 and 4 instead of between 0 and 3. First one is less confusing based on quest_template fields, but second one is more correct because indexing always starts at 0 everywhere.